cloudinit/config/cc_growpart.py

```diff
@@ -93,6 +94,53 @@ class RESIZE(object):
 LOG = logging.getLogger(__name__)
 
 
+@lru_cache()
+def is_lvm_lv(devpath):
+    if util.is_Linux():
+        # all lvm lvs will have a realpath as a 'dm-*' name.
+        rpath = os.path.realpath(devpath)
+        if not os.path.basename(rpath).startswith("dm-"):
+            return False
+        out, _ = subp.subp("udevadm", "info", devpath)
+        # lvs should have DM_LV_NAME=<lvmuuid> and also DM_VG_NAME
+        return 'DM_LV_NAME=' in out
+    else:
+        return False
+
+
+@lru_cache()
+def get_pvs_for_lv(devpath):
+    myenv = {'LANG': 'C'}
+
+    if util.is_Linux():
+        try:
+            (out, _err) = subp.subp(["lvm", "lvs", devpath, "--options=vgname",
+                                     "--noheadings"], update_env=myenv)
+            vgname = out.strip()
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Volume Group information "
+                            "from %s", devpath)
+                raise ResizeFailedException(e) from e
+
+        try:
+            (out, _err) = subp.subp(["lvm", "vgs", vgname, "--options=pvname",
+                                     "--noheadings"], update_env=myenv)
+            pvs = [p.strip() for p in out.splitlines()]
+            if len(pvs) > 1:
+                util.logexc(LOG, "Do not know how to resize multiple Physical"
+                            " Volumes")
+            else:
+                return pvs[0]
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Physical Volume "
+                            "information from Volume Group %s", vgname)
+                raise ResizeFailedException(e) from e
+    else:
+        return False
```

Comment:
Returning False in failure cases versus a "string" in success cases feels confusing for `get_pvs_for_lv`. It looks like behavior like this could cause a problem because we don't check the return value and blindly call `    bname = os.path.basename(rpath) which ultimately leads to a confusing type error in all failure cases here.
` 
I think we probably should emit a warning message like ("No support for get_pvs_for_lv on {system}".format(systemd=platform.system()). If we want to continue with just a warning message, then we probably should be should be checking the return value from get_pvs_for_lv before setting rpath at line 268 `rpath = get_pvs_for_lv(devpath)`

Reply:
I added a check and the warning message :) thanks for the review!